# _****Task 4:****_ <span style=color:yellow;> *****Basic Descriptive Statistics on Dataset*****</span>
<span style=color:white;>**Intern:**</span> *Ibrahim Salman*

<span style=color:white;>**Period:**</span> *20/9/2025 to 10/10/2025*


## <span style=color:yellow;>****_Import Libraries_****</span>

In [ ]:
import numpy as np
import seaborn as sns

## <span style=color:yellow;>****_Load Dataset_****</span>

In [ ]:
df = sns.load_dataset('titanic')
df

survived  pclass     sex   age  sibsp  parch     fare embarked   class  \
0           0       3    male  22.0      1      0   7.2500        S   Third   
1           1       1  female  38.0      1      0  71.2833        C   First   
2           1       3  female  26.0      0      0   7.9250        S   Third   
3           1       1  female  35.0      1      0  53.1000        S   First   
4           0       3    male  35.0      0      0   8.0500        S   Third   
..        ...     ...     ...   ...    ...    ...      ...      ...     ...   
886         0       2    male  27.0      0      0  13.0000        S  Second   
887         1       1  female  19.0      0      0  30.0000        S   First   
888         0       3  female   NaN      1      2  23.4500        S   Third   
889         1       1    male  26.0      0      0  30.0000        C   First   
890         0       3    male  32.0      0      0   7.7500        Q   Third   

       who  adult_male deck  embark_town alive  alone  
0      man        True  NaN  Southampton    no  False  
1    woman       False    C    Cherbourg   yes  False  
2    woman       False  NaN  Southampton   yes   True  
3    woman       False    C  Southampton   yes  False  
4      man        True  NaN  Southampton    no   True  
..     ...         ...  ...          ...   ...    ...  
886    man        True  NaN  Southampton    no   True  
887  woman       False    B  Southampton   yes   True  
888  woman       False  NaN  Southampton    no  False  
889    man        True    C    Cherbourg   yes   True  
890    man        True  NaN   Queenstown    no   True  

[891 rows x 15 columns]

## <span style=color:yellow;>****_Dataset Overview_****</span>

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 15 columns):
 #   Column       Non-Null Count  Dtype   
---  ------       --------------  -----   
 0   survived     891 non-null    int64   
 1   pclass       891 non-null    int64   
 2   sex          891 non-null    object  
 3   age          714 non-null    float64 
 4   sibsp        891 non-null    int64   
 5   parch        891 non-null    int64   
 6   fare         891 non-null    float64 
 7   embarked     889 non-null    object  
 8   class        891 non-null    category
 9   who          891 non-null    object  
 10  adult_male   891 non-null    bool    
 11  deck         203 non-null    category
 12  embark_town  889 non-null    object  
 13  alive        891 non-null    object  
 14  alone        891 non-null    bool    
dtypes: bool(2), category(2), float64(2), int64(4), object(5)
memory usage: 80.7+ KB


## <span style=color:yellow;>****_Check DataTypes_****</span>

In [ ]:
df.dtypes

survived          int64
pclass            int64
sex              object
age             float64
sibsp             int64
parch             int64
fare            float64
embarked         object
class          category
who              object
adult_male         bool
deck           category
embark_town      object
alive            object
alone              bool
dtype: object

## <span style=color:yellow;>****_Check Columns Names_****</span>

In [ ]:
df.columns

Index(['survived', 'pclass', 'sex', 'age', 'sibsp', 'parch', 'fare',
       'embarked', 'class', 'who', 'adult_male', 'deck', 'embark_town',
       'alive', 'alone'],
      dtype='object')

## <span style=color:yellow;>****_Check Dataset Shape_****</span>

In [ ]:
df.shape

(891, 15)

## <span style=color:yellow;>****_Missing Values Handle_****</span>

In [ ]:
df.isnull().sum()

survived         0
pclass           0
sex              0
age            177
sibsp            0
parch            0
fare             0
embarked         2
class            0
who              0
adult_male       0
deck           688
embark_town      2
alive            0
alone            0
dtype: int64

## <span style=color:yellow;>****_Data Cleaning_****</span>

In [213]:
clean = df.drop(columns=['deck'], inplace=True)
clean

In [214]:
df.columns

Index(['survived', 'pclass', 'sex', 'age', 'sibsp', 'parch', 'fare',
       'embarked', 'class', 'who', 'adult_male', 'embark_town', 'alive',
       'alone'],
      dtype='object')

In [215]:
df.shape

(891, 14)

In [216]:
df['age'] = df['age'].fillna(df['age'].mean())
df['age'] 
df['embark_town'] = df['embark_town'].fillna(df['embark_town'].mode()[0])
df['embark_town']
df['embarked'] = df['embarked'].fillna(df['embarked'].mode()[0])
df['embarked']

0      S
1      C
2      S
3      S
4      S
      ..
886    S
887    S
888    S
889    C
890    Q
Name: embarked, Length: 891, dtype: object

## <span style=color:yellow;>****_Dataset After Cleaning_****</span>

In [217]:
df.isnull().sum()

survived       0
pclass         0
sex            0
age            0
sibsp          0
parch          0
fare           0
embarked       0
class          0
who            0
adult_male     0
embark_town    0
alive          0
alone          0
dtype: int64

## <span style=color:yellow;>****_Descriptive Statistics_****</span>

In [ ]:
df.describe()

survived      pclass         age       sibsp       parch        fare
count  891.000000  891.000000  891.000000  891.000000  891.000000  891.000000
mean     0.383838    2.308642   29.699118    0.523008    0.381594   32.204208
std      0.486592    0.836071   13.002015    1.102743    0.806057   49.693429
min      0.000000    1.000000    0.420000    0.000000    0.000000    0.000000
25%      0.000000    2.000000   22.000000    0.000000    0.000000    7.910400
50%      0.000000    3.000000   29.699118    0.000000    0.000000   14.454200
75%      1.000000    3.000000   35.000000    1.000000    0.000000   31.000000
max      1.000000    3.000000   80.000000    8.000000    6.000000  512.329200

## <span style=color:yellow;>****_Description of Statistical Analysis_****</span>

#### <span style=color:yellow;>****_Key Insights_****</span>


1. Survival Rate: Only 38.4% of passengers in the dataset survived

2. Passenger Class: Most passengers (75%) were in 3rd class

3. Age Distribution: Average age was 29.7 years with a wide range (0.42 to 80 years)

4. Family Size: Most passengers traveled alone (median SibSp and Parch = 0)

5. Fare Distribution: Highly skewed with most fares under $31 but a maximum of $512.33



#### <span style=color:yellow;>****_Missing Values Handling_****</span>


1. Successfully handled missing values in 'age' (177), 'embarked' (2), and 'embark_town' (2)

2. Removed the 'deck' column with 688 missing values (77% missing)

3. Used appropriate imputation strategies:

   - Age: Mean imputation

   - Embarked/Embark_town: Mode imputation

This analysis provides a solid foundation for further exploratory data analysis and predictive modeling on the Titanic dataset.